File: lib/ocaml/dom_storage.ml
type t = Dom_storage2.t

external getItem : string -> string option = "getItem" [@@bs.send.pipe: t] [@@bs.return null_to_opt]
external setItem : string -> string -> unit = "setItem" [@@bs.send.pipe: t]
external removeItem : string -> unit = "removeItem" [@@bs.send.pipe: t]
external clear : unit = "clear" [@@bs.send.pipe: t]
external key : int -> string option = "key" [@@bs.send.pipe: t] [@@bs.return null_to_opt]
external length : t -> int = "length" [@@bs.get]

external localStorage : t = "localStorage" [@@bs.val]
external sessionStorage : t = "sessionStorage" [@@bs.val]

File: lib/js/Dom_storage.js
'use strict';

let Primitive_option = require("./Primitive_option.js");

function getItem(s, obj) {
  return Primitive_option.fromNull(obj.getItem(s));
}

function setItem(k, v, obj) {
  obj.setItem(k, v);
}

function removeItem(s, obj) {
  obj.removeItem(s);
}

function key(i, obj) {
  return Primitive_option.fromNull(obj.key(i));
}

exports.getItem = getItem;
exports.setItem = setItem;
exports.removeItem = removeItem;
exports.key = key;
/* No side effect */
